client/src/tools/SearchBar.js

```diff
@@ -0,0 +1,49 @@
+import SearchIcon from '@material-ui/icons/Search';
+import CloseIcon from '@material-ui/icons/Close';
+import {useState} from 'react';
+import styles from '../css/searchBar.module.css';
+
+function SearchBar () {
+
+    // const [filteredData, setFilteredData] = useState([]);
+    const [wordEntered, setWordEntered] = useState("");  
+    const [isSearching, setIsSearching] = useState(false);
+
+    const handleSearch = (e) => {
+        const searchWord = e.target.value;
+        setWordEntered(searchWord);
+    
+        if (searchWord === "") {
+        //   setFilteredData([]);
+            setIsSearching(false);
+        } else {
+        //   setFilteredData(searchWord);
+          setIsSearching(true);
+        }
+    }
+    
+    const clearInput = () => {
+        // setFilteredData([]);
+        setWordEntered("");
+        setIsSearching(false);
+    };
+    
+    const BarIcon = isSearching ? CloseIcon : SearchIcon;
+
+    return (
+        <div className={styles.search}>
+            <input 
+                type="text" 
+                className={styles.inputArea && styles.searchTerm}
```

Comment:
For variable like `styles.inputArea`, the value is actually a string (you can console.log to check), so in JavaScript when you use AND operator with strings, it returns the last string:

```java
const s1 = 'foo';
const s2 = 'bar';

console.log(s1 && s2); // bar
```
Which I think is not the purpose of this line of code